src/qComponents/QInputNumberNew/src/QInputNumberNew.vue

```diff
@@ -0,0 +1,242 @@
+    const qFormItem = inject<Nullable<QFormItemProvider>>('qFormItem', null);
+    const inputRef = ref<Nullable<HTMLInputElement>>(null);
+
+    const precision = computed<number>(() => Math.max(props.precision ?? 0, 0));
+
+    const isDisabled = computed<boolean>(
+      () => props.disabled || (qForm?.disabled.value ?? false)
+    );
+
+    let internalValue: Nullable<string> = null;
+
+    const changesEmitter = (type: 'input' | 'change', value: string): void => {
+      const emittedValue =
+        value && !Number.isNaN(Number(value)) ? Number(value) : null;
+
+      if (props.modelValue === emittedValue) return;
+
+      ctx.emit('update:modelValue', emittedValue);
+      ctx.emit(type, emittedValue);
+      if (props.validateEvent) qFormItem?.validateField(type);
+    };
+
+    const getValueMatchRegExp = ({
+      matchEnd
+    }: {
+      matchEnd: boolean;
+    }): RegExp => {
+      const signPattern = (props.min ?? MIN_INTEGER) < 0 ? '-?' : '';
+      const integerPattern = '[0-9]*';
+      const fractionPattern =
+        precision.value > 0 ? `[.,]?[0-9]{0,${precision.value}}` : '';
+      const endPattern = matchEnd ? '$' : '';
+
+      return new RegExp(
+        `^${signPattern}${integerPattern}${fractionPattern}${endPattern}`
+      );
+    };
+
+    const matchValue = (value: string): string => {
+      const cleanValue = value
+        .replace(/[^0-9,.-]/g, '')
+        .replace(/-{2,}/g, '-')
+        .replace(',', '.');
+      const match = cleanValue.match(getValueMatchRegExp({ matchEnd: false }));
+
+      return cleanValue && match ? String(Number(match[0])) : '';
+    };
+
+    const testValue = (value: string): boolean => {
+      const hasIncorrectCombination = /^(--|(-?(0{2,}|0[1-9])))/.test(value);
+
+      return (
+        !hasIncorrectCombination &&
+        getValueMatchRegExp({ matchEnd: true }).test(value)
+      );
+    };
+
+    const getClampedValue = (value: string): string => {
+      const min = props.min ?? MIN_INTEGER;
+      const max = props.max ?? MAX_INTEGER;
+
+      if (!value) return '';
+      if (Number(value) < min) return String(min);
+      if (Number(value) > max) return String(max);
+      return value;
+    };
+
+    const handlePaste = (event: InputEvent): void => {
+      const target = event.target as HTMLInputElement;
+
+      internalValue = getClampedValue(matchValue(target.value));
+      target.value = internalValue;
+    };
+
+    const handleInput = (event: InputEvent): void => {
+      const target = event.target as HTMLInputElement;
+
+      if (testValue(target.value)) {
+        internalValue = getClampedValue(target.value);
+      }
+      target.value = internalValue ?? '';
+    };
+
+    const handleInputCommon = (event: InputEvent): void => {
+      const target = event.target as HTMLInputElement;
+
+      if (event.inputType === 'insertFromPaste') {
+        handlePaste(event);
+      } else {
+        handleInput(event);
```

Comment:
```suggestion
        handlePaste(event);
      } else {
        handleInput(event);
```

давай вернём логики раз тут отрефачилось так

Reply:
@ViZhe вернул